File: scripts/test_mcp_nuxt.py
import requests
import sseclient
import json
import threading
import time

def test_nuxt_mcp():
    url = "http://localhost:3000/__mcp/sse"
    print(f"Connecting to SSE: {url}")
    
    response = requests.get(url, stream=True)
    client = sseclient.SSEClient(response)
    
    session_id = None
    endpoint = None
    
    for event in client.events():
        print(f"Received event: {event.event}")
        if event.event == 'endpoint':
            endpoint = event.data
            # Extract session ID if needed, or just use the endpoint provided
            print(f"Endpoint: {endpoint}")
            
            # Now we can send the initialize request
            full_post_url = f"http://localhost:3000{endpoint}"
            print(f"POST URL: {full_post_url}")
            
            # 1. Initialize
            init_payload = {
                "jsonrpc": "2.0",
                "method": "initialize",
                "params": {
                    "protocolVersion": "2024-11-05",
                    "capabilities": {},
                    "clientInfo": {"name": "manual-test", "version": "1.0"}
                },
                "id": 1
            }
            res = requests.post(full_post_url, json=init_payload)
            print(f"Initialize Response: {res.status_code} - {res.text}")
            
            # 2. List Tools
            tools_payload = {
                "jsonrpc": "2.0",
                "method": "tools/list",
                "params": {},
                "id": 2
            }
            res = requests.post(full_post_url, json=tools_payload)
            print(f"Tools Response: {res.status_code} - {res.text}")
            
            break # Success, we can stop

if __name__ == "__main__":
    test_nuxt_mcp()

File: scripts/test_mcp_nuxt.js
async function testNuxtMcp() {
  const url = "http://localhost:3000/__mcp/sse";
  console.log(`Connecting to SSE: ${url}`);

  const controller = new AbortController();
  const response = await fetch(url, { signal: controller.signal });

  if (!response.ok) {
    console.error(
      `Failed to connect: ${response.status} ${response.statusText}`,
    );
    return;
  }

  const reader = response.body.getReader();
  const decoder = new TextDecoder();
  let buffer = "";
  let initialized = false;

  // Process stream
  (async () => {
    while (true) {
      const { value, done } = await reader.read();
      if (done) break;

      buffer += decoder.decode(value, { stream: true });
      const lines = buffer.split("\n\n");
      buffer = lines.pop();

      for (const block of lines) {
        const eventLine = block.match(/event: (.*)/);
        const dataLine = block.match(/data: (.*)/);

        if (eventLine && dataLine) {
          const event = eventLine[1].trim();
          const data = dataLine[1].trim();

          if (event === "endpoint") {
            console.log(`[SSE] Endpoint received: ${data}`);
            const postUrl = `http://localhost:3000${data}`;

            // Send Initialize
            console.log("[POST] Sending initialize...");
            await fetch(postUrl, {
              method: "POST",
              headers: { "Content-Type": "application/json" },
              body: JSON.stringify({
                jsonrpc: "2.0",
                method: "initialize",
                params: {
                  protocolVersion: "2024-11-05",
                  capabilities: {},
                  clientInfo: { name: "manual-test", version: "1.0" },
                },
                id: 1,
              }),
            });
          } else if (event === "message") {
            console.log(`[SSE] Message received: ${data}`);
            const msg = JSON.parse(data);

            if (msg.id === 1) {
              console.log("✅ Initialize success!");
              const postUrl = `http://localhost:3000/__mcp/messages?sessionId=${msg.sessionId || "unknown"}`;
              // Note: sessionId might not be in the msg, but was in endpoint.
              // Actually we should assume the postUrl is constant or use the one from endpoint.
              // But let's just trigger the tools list now.

              // Re-parse endpoint from log? No, let's just reuse the logic.
              // Simplified: Just waiting for ID 1 to confirm connection works.

              console.log("✅ Nuxt MCP is working!");
              process.exit(0);
            }
          }
        }
      }
    }
  })();
}

testNuxtMcp().catch(console.error);
